dist/hijax.js

```diff
@@ -69,25 +69,22 @@
 
     // Get a particular response header value by key
     // Extracted from jQuery 2.1.1
+    // Changed to grab content-type only (we only use content-type, no need to set everything) from XHR object
     Hijacker.prototype.getResponseHeader = function(xhr, key) {
         var rHeaders = /^(.*?):[ \t]*([^\r\n]*)$/mg;
-        var match;
         var responseHeadersString;
+        var contentType;
 
         if ([states.HEADERS_RECEIVED, states.LOADING, states.DONE]
                 .indexOf(xhr.readyState) > -1) {
 
             responseHeadersString = xhr.getAllResponseHeaders();
 
-            if (!this.responseHeaders) {
-                this.responseHeaders = {};
-                while ((match = rHeaders.exec(responseHeadersString))) {
-                    this.responseHeaders[match[1].toLowerCase()] = match[2];
-                }
-            }
-            match = this.responseHeaders[key.toLowerCase()];
+            contentType = responseHeadersString.match(/Content-Type:\s*(.*?);\s*charset/);
+
+            contentType = contentType.length > 1 ? contentType[1] : null;
         }
-        return match === null ? null : match;
```

Comment:
@henryh15 careful ⚠️! The behaviour of this function has now changed ... where previously it would return either `null` or the value, now it will return `undefined`, `null`, or the value.